irohad/ordering/impl/on_demand_ordering_gate.cpp

```diff
@@ -147,31 +148,49 @@ OnDemandOrderingGate::removeReplays(
         });
   };
 
+  std::unordered_set<std::string> hashes;
+  auto tx_is_not_repeat = [&hashes](const auto &tx) {
+    auto tx_hash = tx.hash().hex();
+
+    if (hashes.count(tx_hash)) {
+      return false;
+    } else {
+      hashes.insert(tx_hash);
+      return true;
+    }
+  };
+
   shared_model::interface::TransactionBatchParserImpl batch_parser;
 
   bool has_replays = false;
+  bool has_repetition = false;
   auto batches = batch_parser.parseBatches(proposal->transactions());
   for (auto &batch : batches) {
     bool all_txs_are_new =
         std::all_of(batch.begin(), batch.end(), tx_is_not_processed);
+    bool no_duplicates_transaction =
+        std::all_of(batch.begin(), batch.end(), tx_is_not_repeat);
```

Comment:
It seems like you can use single `std::unordered_set` instead of two checks in a loop via `std::all_of`.